util/network-devp2p/src/discovery.rs

```diff
@@ -95,10 +121,14 @@ pub struct Discovery {
 	discovery_id: NodeId,
 	discovery_nodes: HashSet<NodeId>,
 	node_buckets: Vec<NodeBucket>,
+	in_flight_requests: HashMap<NodeId, PendingRequest>,
+	expiring_pings: VecDeque<(NodeId, Instant)>,
+	expiring_finds: VecDeque<(NodeId, Instant)>,
 	send_queue: VecDeque<Datagramm>,
 	check_timestamps: bool,
 	adding_nodes: Vec<NodeEntry>,
 	ip_filter: IpFilter,
+	request_backoff: Vec<Duration>,
```

Comment:
> It's on Discovery, not NodeBucket.

Of course, sorry. Typo.

>  And the motivation is just injecting dependencies explicitly.

I was just thinking that since `Discovery::new` refers to `const REQUEST_BACKOFF` anyway only to map its integers to `Duration`s, why not declare `REQUEST_BACKOFF` itself as a `&'static [Duration]` and only assign the reference to `Duration::request_backoff`. After all, client code can not change the backoff setting and in test modules one can still define other backoff sequences and overwrite the default. But it was just a quick thought and certainly not important.

Reply:
Added 364c8e1001c367e45693459d0002c9eacd9f3621. Let me know if you prefer that or want me to revert. Shouldn't make any difference performance-wise because Discovery is only instantiated once.